src/main/scala/mesosphere/marathon/core/task/bus/MarathonTaskStatus.scala

```diff
@@ -31,12 +32,20 @@ object MarathonTaskStatus {
   case class Starting(mesosStatus: Option[TaskStatus]) extends MarathonTaskStatus
   case class Running(mesosStatus: Option[TaskStatus]) extends MarathonTaskStatus
 
+  // Note that we need to distinguish between TemporarilyUnreachable tasks and really lost tasks that won't come back
+  // tl;dr: don't use Lost
+  case class Lost(mesosStatus: Option[TaskStatus]) extends MarathonTaskStatus
```

Comment:
Actually not, but if we removed it we'd be unable to create an object in the MarathonTaskStatus.apply:
> `case TASK_LOST     => Lost`

Basically, I only moved this from few lines below because it doesn't extend Terminal any longer. I would consider removing `MarathonTaskStatus` but not in the .15 branch, okay?